### PR TITLE
Add Test for Replication Eventing - ObjC

### PR DESCRIPTION
**Pull request by jayahariv**
* add two unit tests for replication eventing.
* expiration with push and pull.

link: #2264

### CODE REVIEW

File: Objective-C/Tests/ReplicatorTest.m [hunk 1]
```diff
@@ -1773,6 +1773,91 @@ - (void) testPullFilter {
...
 
+- (void) testReplicationEventExpirationWithPush {
+    NSError* error;
+    CBLMutableDocument* doc = [[CBLMutableDocument alloc] initWithID: @"doc"];
+    [doc setString: @"Tiger" forKey: @"species"];
+    [doc setString: @"Hobbes" forKey: @"pattern"];
+    Assert([self.db saveDocument: doc error: &error]);
+    
+    XCTestExpectation* expectation = [self expectationWithDescription: @"Document expiry test"];
+    id token1 = [self.db addDocumentChangeListenerWithID: doc.id
```
Comment (jayahariv) [inline Objective-C/Tests/ReplicatorTest.m]: will change the variable name from token1 to meaningful name, `documentChangeToken`.

File: Objective-C/Tests/ReplicatorTest.m [hunk 2]
```diff
@@ -1773,6 +1773,91 @@ - (void) testPullFilter {
...
+    [self waitForExpectationsWithTimeout: 5.0 handler: nil];
+    
+    AssertEqual(self.db.count, 0u);
+    AssertEqual(otherDB.count, 1u);
+    [self.db removeChangeListenerWithToken: token1];
+    [replicator removeChangeListenerWithToken: token];
+}
+
+
+- (void) testReplicationEventExpirationWithPull {
```
Comment (pasin) [inline Objective-C/Tests/ReplicatorTest.m]: I don't see the value of this test. Is this the same thing that .NET is testing?

Comment (pasin) [reply Objective-C/Tests/ReplicatorTest.m]: Test 5th is not valid for CBL. We don't have purge and become tombstone. What .NET has is test 4th which is testReplicationEventExpirationWithPush.

Can you rename the test `testReplicationEventExpirationWithPush` to `testPushAndForget`?

Comment (pasin) [reply Objective-C/Tests/ReplicatorTest.m]: I suggested removing testReplicationEventExpirationWithPull.

Comment (jayahariv) [reply Objective-C/Tests/ReplicatorTest.m]: Yes, will correct that.

**Pull request merged**